pkg/util/chunkinfologger/chunkinfologger.go

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkinfologger
+
+import (
+	"fmt"
+	"hash/crc32"
+	"strconv"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/mimir/pkg/ingester/client"
+	"github.com/grafana/mimir/pkg/storegateway/storepb"
+)
+
+// Allow for 10% overhead from string quotes when logger dumps the JSON.
+const maxSize = (64 * 1024 * 90) / 100
+
+type ChunkInfoLogger struct {
+	chunkInfo strings.Builder
+	msg       string
+	traceID   string
```

Comment:
Should we include the span ID as well? In a query with multiple selectors (or a query that gets sharded), the span should help us work out which log lines belong to a single set of ingester or store-gateway requests.

Reply:
I haven't run into that yet, but a quick look at the logs tells me that some continuous test queries are indeed sharded. I'll add it.

Reply:
Added via dskit